File: clients/node/examples/find.js
import { Screen, Pattern } from "../src";

async function main() {
  const screen = await Screen();
  try {
    const pattern = Pattern("assets/pattern.png").exact();
    const match = await screen.find(pattern);
    console.log(`found rect=(${match.x},${match.y},${match.w},${match.h}) target=(${match.targetX},${match.targetY})`);
  } finally {
    await screen.close();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: clients/python/examples/find.py
from __future__ import annotations

from sikuligo import Pattern, Screen


def main() -> int:
    screen = Screen()
    try:
        match = screen.find(Pattern("assets/pattern.png").exact(), timeout_millis=3000)
        print(
            f"match rect=({match.x},{match.y},{match.w},{match.h}) "
            f"score={match.score:.3f} target=({match.target_x},{match.target_y})"
        )
    finally:
        screen.close()
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
